core/src/main/java/org/elasticsearch/index/query/MoreLikeThisQueryBuilder.java

```diff
@@ -147,7 +204,8 @@ public Item doc(BytesReference doc) {
          * Sets to a given artificial document, that is a document that is not present in the index.
          */
         public Item doc(XContentBuilder doc) {
-            return this.doc(doc.bytes());
+            this.doc = doc != null ? doc.bytes() : null;
```

Comment:
In fact, given that we require either doc or an index document id, this should really be a constructor and this setter should be removed. That way the user can only specify doc OR id and must specify one of them. We should add null checks in the constructors and throw IllegalArgumentException if nulls are passed in